src/ir/moduledef.cpp

```diff
@@ -223,6 +223,20 @@ void ModuleDef::connect(std::initializer_list<const char*> pA, std::initializer_
 void ModuleDef::connect(std::initializer_list<std::string> pA, std::initializer_list<string> pB) {
   connect(SelectPath(pA.begin(),pA.end()),SelectPath(pB.begin(),pB.end()));
 }
+bool ModuleDef::hasConnection(Wireable* a, Wireable* b) {
+  auto sorted = std::minmax(a,b);
+  Connection con(sorted.first,sorted.second);
+  return connections.count(con) > 0;
+}
+
+Connection ModuleDef::getConnection(Wireable* a, Wireable* b) {
+  auto sorted = std::minmax(a,b);
+  Connection con(sorted.first,sorted.second);
+  if (connections.count(con) > 0) {
+    return *connections.find(con);
+  }
+  return con;
+}
```

Comment:
If I understand this correctly, it will return a new connection if it's not present in the ModuleDef, but won't add it to the `connections` data structure. That seems confusing, is there a reason for that?

The behavior I would expect is some sort of error or null value if the connection isn't present, but perhaps this "default" connection behavior is useful somewhere else?